### PR TITLE
misc: Update README and devcontainer to include GPU-FS commands, docker image

### PR DESCRIPTION
**Pull request by v-ramadas**
Adds commands to run GPU-FS and to take a checkpoint/restore it to the README

### CODE REVIEW

**Comment by mattsinc** [comment]
@powerjg : just making sure you see this.

**Review by mattsinc — COMMENTED** [review]
From my perspective this looks ok, but will wait for @powerjg

**Comment by powerjg** [comment]
I did not see this. I'll review now... I think there are some misunderstandings

**Comment by v-ramadas** [comment]
I'm re-running all the commands I listed in the README from a new codespace. I found one other typo that I'm fixing. I will ping here once I confirm all of them work

File: README.md [hunk 1]
```diff
@@ -128,3 +128,40 @@ Below is an example command
 ```sh
 docker run -v $PWD:$PWD -v /usr/local/bin:/usr/local/bin -w $PWD ghcr.io/gem5/gcn-gpu:v24-0 gem5-vega-se gem5/configs/example/apu_se.py -n 3 -c square
 ```
+
+## Note on running GPU FS
+
+You need not use docker to run simulations in GPU FS mode. You still need docker to compile applications like square. We will be simulating AMD's latest GPU, MI300, in FS mode
```
Comment (powerjg) [inline README.md:134]: Why do you need a docker here? I've installed rocm in this container. See https://github.com/gem5/gem5/pull/1282/files#diff-4a46b3d1ca68a5308768c3ae88fd7bb425557a9a63cab90840b6c8c73e0473d3R71

File: README.md [hunk 2]
```diff
@@ -128,3 +128,40 @@ Below is an example command
...
+Below is an example command
+
+```sh
+# To compile square
+cd /workspaces/gem5-bootcamp-env
+wget https://cs.wisc.edu/~ramadas/gem5-isca24/gpu-fs/Makefile
```
Comment (powerjg) [inline README.md:140]: Is there a reason you have to wget these things? Why not just copy them into materials/isca24

File: README.md [hunk 3]
```diff
@@ -128,3 +128,40 @@ Below is an example command
...
 docker run -v $PWD:$PWD -v /usr/local/bin:/usr/local/bin -w $PWD ghcr.io/gem5/gcn-gpu:v24-0 gem5-vega-se gem5/configs/example/apu_se.py -n 3 -c square
 ```
+
+## Note on running GPU FS
+
+You need not use docker to run simulations in GPU FS mode. You still need docker to compile applications like square. We will be simulating AMD's latest GPU, MI300, in FS mode
+Below is an example command
+
+```sh
+# To compile square
```
Comment (powerjg) [inline README.md:138]: I would put this in materials/isca24/10-gpufs/README.md, not in the base readme file.

Or, you could just have it in your slides.

File: README.md (hunk 1 above)
Comment (v-ramadas) [reply README.md:134]: My bad...I did not see this. Will change it now

File: README.md (hunk 2 above)
Comment (v-ramadas) [reply README.md:140]: Sounds good. Will do that.

File: README.md (hunk 3 above)
Comment (v-ramadas) [reply README.md:138]: Will move it

**Comment by v-ramadas** [comment]
I've created a new pull request with the changes. I will close this one

**Pull request closed without merging**